Sources/NetworkService.swift

```diff
@@ -19,80 +19,109 @@ struct NetworkService: PushNotificationsNetworkable {
+    func unsubscribeAll(completion: @escaping CompletionHandler) {
         self.setSubscriptions(interests: [], completion: completion)
     }
 
-    func track(userInfo: [AnyHashable: Any], eventType: String, deviceId: String, completion: @escaping () -> Void = {}) {
+    func track(userInfo: [AnyHashable: Any], eventType: String, deviceId: String, completion: @escaping CompletionHandler) {
         guard let publishId = PublishId(userInfo: userInfo).id else { return }
         let timestampSecs = UInt(Date().timeIntervalSince1970)
         guard let body = try? Track(publishId: publishId, timestampSecs: timestampSecs, eventType: eventType, deviceId: deviceId).encode() else { return }
 
         let request = self.setRequest(url: self.url, httpMethod: .POST, body: body)
         self.networkRequest(request, session: self.session) { (response) in
-            completion()
+            switch response {
+            case .Success(_, let httpURLResponse):
+                completion(nil, httpURLResponse)
+            case .Failure(_, let httpURLResponse):
+                completion(nil, httpURLResponse)
+            }
         }
     }
 
-    func syncMetadata(completion: @escaping () -> Void = {}) {
+    func syncMetadata(completion: @escaping CompletionHandler) {
         guard let metadataDictionary = Metadata.load() else { return }
         let metadata = Metadata(propertyListRepresentation: metadataDictionary)
         if metadata.hasChanged() {
             let updatedMetadataObject = Metadata.update()
             guard let body = try? updatedMetadataObject.encode() else { return }
             let request = self.setRequest(url: self.url, httpMethod: .PUT, body: body)
             self.networkRequest(request, session: self.session) { (response) in
-                completion()
+                switch response {
+                case .Success(_, let httpURLResponse):
+                    completion(nil, httpURLResponse)
+                case .Failure(_, let httpURLResponse):
+                    completion(nil, httpURLResponse)
+                }
             }
         }
     }
 
     // MARK: Networking Layer
     private func networkRequest(_ request: URLRequest, session: URLSession, completion: @escaping NetworkCompletionHandler) {
         session.dataTask(with: request, completionHandler: { (data, response, error) in
-            guard let data = data else { return }
             guard
-                let httpURLResponse = response as? HTTPURLResponse, httpURLResponse.statusCode == 200, error == nil
-            else {
-                guard let reason = try? JSONDecoder().decode(Reason.self, from: data) else { return }
-                return completion(NetworkResponse.Failure(description: reason.description))
+                let data = data,
+                let httpURLResponse = response as? HTTPURLResponse
+            else { return }
+
+            guard httpURLResponse.statusCode == 200, error == nil else {
```

Comment:
Just in case in the future we want endpoints to return a more precise status code: `httpURLResponse.statusCode >= 200 && httpURLResponse.statusCode < 300`